src/utils/toc.ts

```diff
@@ -0,0 +1,41 @@
+import slugger from 'github-slugger';
+
+interface ToCItem {
+  text: string;
+  id: string;
+  level: string;
+  subItems: ToCItem[];
+}
+
+export type TableOfContents = ToCItem[];
+
+export function getTableOfContents(mdxContent: string) {
```

Comment:
The `x` might be misleading as this function is not restricted to mdx content.

```suggestion
export function getTableOfContents(mdContent: string) {
```